redash/models.py

```diff
@@ -945,6 +1150,13 @@ def recent(cls, org, groups, user_id, for_user=False, limit=20):
     def get_by_slug_and_org(cls, slug, org):
         return cls.get(cls.slug == slug, cls.org==org)
 
+    def tracked_save(self, changing_user, old_object=None, *args, **kwargs):
+        self.version += 1
+        self.save(*args, **kwargs)
+        # save Change record
+        new_change = Change.save_change(user=changing_user, old_object=old_object, new_object=self)
+        return new_change
```

Comment:
Also I just noticed that you didn't implement the actual optimistic locking here...